core/store/src/main/java/uk/gov/gchq/gaffer/store/schema/SchemaElementDefinitionValidator.java

```diff
@@ -168,10 +166,10 @@ protected ValidationResult validateFunctionArgumentTypes(
         return result;
     }
 
-    private ValidationResult validateDirection(final SchemaElementDefinition elementDef) {
+    protected ValidationResult validateDirection(final SchemaElementDefinition elementDef) {
         final ValidationResult result = new ValidationResult();
         if (elementDef instanceof SchemaEdgeDefinition && elementDef.getIdentifierTypeName(IdentifierType.DIRECTED) == null) {
-            result.addError("\"directed\" must be set");
+            LOGGER.warn("Edge definition does not have directed set.");
```

Comment:
Within a definition you cannot know the name of the group that is using the definition (annoyingly). This is because there is a Map of group names to definitions.